### PR TITLE
Skills adding + search

### LINKED ISSUE #56: Create skills model

### PR DESCRIPTION
**Pull request by MasterCoookie**
closes #56 #55

### CODE REVIEW

File: models/skillsModel.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const mongoose = require('mongoose');
+
+const skillsSchema = new mongoose.Schema({
+    name: {
+        type: String,
+        required: true,
+        unique: true,
+        minlength: [2, 'Name must be at least 2 characters long'],
```
Comment (WojciechPtas) [inline models/skillsModel.js:8]: I propose to change this value to at least three

File: controllers/skillsController.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            console.log('Error creating skill');
+            console.log(e);
+        }
+        res.status(400).json({ errors });
+    }
+}
+
+const skillsCreator_get = (req, res) => {
+    try {
+        res.render('skills/skillsCreator', { title: 'Create skill' });
```
Comment (macibol775) [inline controllers/skillsController.js:31]: Create a skill OR Create skills

File: public/js/skillsSearch.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    if(skills.length == 0) {
+                        searchResultsDOMElement.innerHTML = 'No skills found';
+                        return;
+                    }
+                    searchResultsDOMElement.innerHTML = '';
+                    skills.forEach(skill => {
+                        const skillDOMElement = document.createElement('div');
+                        skillDOMElement.classList.add('skill');
+                        skillDOMElement.innerHTML = skill.name;
+                        skillDOMElement.addEventListener('click', function () {
```
Comment (WojciechPtas) [inline public/js/skillsSearch.js:33]: is the keyword `function` necessary here? It seems like we dont use it in other parts of the project while addin listeners

File: public/js/skillsSearch.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+window.addEventListener('load', function () {
```
Comment (WojciechPtas) [inline public/js/skillsSearch.js:1]: As mentioned below

**Review by WojciechPtas — APPROVED** [review]
LGTM

File: views/partials/skillsComponent.ejs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<div id="skillsComponent">
+    <input type="text" id="skillsSearchBar" />
+    <div id="skillsResults" class=" border h-100">
```
Comment (macibol775) [inline views/partials/skillsComponent.ejs:3]: space in 'class=" border...' should be removed

**Review by macibol775 — APPROVED** [review]

**Pull request merged**